tools/mkmodel/mkmodel.c

```diff
@@ -1154,15 +1260,121 @@ void sample_anim_channel(gltf_anim_channel_t *channel, float time, float *out)
     }
 }
 
-void make_anim_channel_samples(gltf_anim_channel_t *channel, float fps, uint32_t num_frames)
+void add_anim_sample(gltf_keyframe_samples_t *samples, float time, float *data, float num_components)
+{
+    samples->num_keyframes++;
+    samples->keyframes = realloc(samples->keyframes, sizeof(gltf_keyframe_t)*samples->num_keyframes);
+    memcpy(&samples->keyframes[samples->num_keyframes-1].data, data, num_components*sizeof(float)); 
+    samples->keyframes[samples->num_keyframes-1].time = time;
+}
+
+void remove_anim_sample(gltf_keyframe_samples_t *samples, size_t index)
+{
+    if(index >= samples->num_keyframes-1) {
+        samples->num_keyframes--;
+        return;
+    }
+    memmove(&samples->keyframes[index], &samples->keyframes[index+1], (samples->num_keyframes-index-1)*sizeof(gltf_keyframe_t)); 
+    samples->num_keyframes--;
+}
+
+float catmull_calc(float p1, float p2, float p3, float p4, float t)
+{
+    float a = (-1*p1  +3*p2 -3*p3 + 1*p4) * t*t*t;
+    float b = ( 2*p1  -5*p2 +4*p3 - 1*p4) * t*t;
+    float c = (  -p1        +  p3)        * t;
+    float d = 2*p2;
+    return 0.5f * (a+b+c+d);
+}
+
+void catmull_calc_vec(float *p1, float *p2, float *p3, float *p4, float *out, float t, int num_values)
+{
+    for(int i=0; i<num_values; i++) {
+        out[i] = catmull_calc(p1[i], p2[i], p3[i], p4[i], t);
+    }
+}
```

Comment:
These are duplicated from model64.c, right? Maybe put them in a shared header?

Reply:
Did that